ruma-api-macros/src/api/metadata.rs

```diff
@@ -139,12 +160,21 @@ enum FieldValue {
     Method(Ident),
     Name(LitStr),
     Path(LitStr),
-    RateLimited(LitBool),
-    Authentication(Ident),
+    RateLimited(LitBool, Vec<Attribute>),
+    Authentication(Ident, Vec<Attribute>),
 }
 
 impl Parse for FieldValue {
     fn parse(input: ParseStream) -> syn::Result<Self> {
+        let attrs: Vec<Attribute> = input.call(Attribute::parse_outer)?;
+        for attr in attrs.iter() {
+            if !util::is_cfg_attribute(attr) {
+                return Err(syn::Error::new_spanned(
+                    &attr,
+                    "attributes may only be `cfg` attributes",
```

Comment:
The way this is currently worded it sounds like a general statement, not specific to the `metadata` block. This is only a suggestion but I think including "here" in the message would be good.

```suggestion
                    "only `cfg` attributes may appear here",
```

Reply:
> The way this is currently worded it sounds like a general statement

Agreed, using the word "attributes" twice didn't sit well with me either.